sway-core/src/error.rs

```diff
@@ -29,6 +29,14 @@ macro_rules! check {
     }};
 }
 
+macro_rules! append {
```

Comment:
How come we're diverging from the usual `CompileResult` for methods like `unify_with_self`? Can we not achieve the same behavior using `CompileResult<()>` and `check!`?

Reply:
No significant reason, `unify_with_self` returns `(Vec<CompileWarning>, Vec<TypeError>)` with no value, while `CompileResult` needs a `Vec<CompileError>` and does return a value. It would be simple to make `unify_with_self` return a `CompileResult<()>` though.

Reply:
Yeah that's fine I suppose. Looks good as is!